### PR TITLE
done Level9

### PR DESCRIPTION
**Pull request by ElizavetaSergeeva-Nekrasova**

### CODE REVIEW

File: src/MechanicService.java [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+
+    private static int getRandomArrayElement() {
+        return (int) (Math.random() * NUMBER_OF_DETAILS);
+    }
+
+    private static int getRandomNumberOfBreaks() {
+        return (int) ((Math.random() * MAX_NUMBER_OF_BREAKS) + 1);
+    }
+
+    private static void writeToFile(Vehicle vehicle, Map<String, Integer> map) {
```
Comment (AlexeyKatomachin) [inline src/MechanicService.java:78]: How much activities do this method?

Comment (ElizavetaSergeeva-Nekrasova) [reply src/MechanicService.java:78]: исправила

**Review by AlexeyKatomachin — APPROVED** [review]

File: src/Main.java [hunk 2]
```diff
@@ -35,6 +31,67 @@ private static void showVehicleArray() {
...
+    private static int getBreaksCount(Map<String, Integer> map ) {
+        int numberOfBreaks = 0;
+
+        if (map.isEmpty()) {
+            return 0;
+        } else {
+            for (Map.Entry<String, Integer> entry:
+                    map.entrySet()) {
+                numberOfBreaks += entry.getValue();
+            }
```
Comment (nikoteressi) [inline src/Main.java]: The 'else' construct is unnecessary here. 

When 'if' is true, 0 will be returned and the method will terminate.

File: src/MechanicService.java [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+    @Override
+    public Map<String, Integer> detectBreaking(Vehicle vehicle) {
+        Map<String, Integer> map = new HashMap<>();
+        int numberOfBrokenDetails = getRandomNumberOfBrokenDetails();
+
+        int i = 0;
+        while (i < numberOfBrokenDetails) {
+            String detail = DETAILS[getRandomArrayElement()];
+            int numberOfBreaks = getRandomNumberOfBreaks();
+            map.put(detail, numberOfBreaks);
```
Comment (nikoteressi) [inline src/MechanicService.java]: This block can be extracted to a separate method. For example: 'setMapOfBrokenParts'

File: src/Main.java (hunk 2 above)
Comment (ElizavetaSergeeva-Nekrasova) [reply src/Main.java]: fixed

File: src/MechanicService.java (hunk 3 above)
Comment (ElizavetaSergeeva-Nekrasova) [reply src/MechanicService.java]: fixed

**Review by nikoteressi — APPROVED** [review]

**Pull request merged**